file.go

```diff
@@ -698,22 +701,32 @@ func (f *filePages) columnPath() columnPath {
 }
 
 var (
-	bufioReaderPool sync.Pool
+	bufioReaderPool = map[int]*sync.Pool{}
```

Comment:
We need access to this map to be synchronized on a mutex, or use a `sync.Map` to prevent race conditions when multiple goroutines are using the map.

Reply:
oof, that seems obvious in retrospect. thank you.